src/main/java/duke/command/CommandList.java

```diff
@@ -0,0 +1,152 @@
+        Todo newTodo = new Todo(description);
+        items.add(taskCount, newTodo);
+    }
+    public void removeItem(ArrayList<Task> items, int taskNum) {
+        taskCount--;
+        removeTaskMessage(items.get(taskNum));
+        items.remove(taskNum);
+    }
+
+    public static void printEndMessage() {
+        System.out.println(border);
+        System.out.println("chat again next time!\n" + border);
+    }
+
+    public void printErrorMessage(DukeException error) {
+        error.printError(error.getErrorType());
+    }
+
+    public void executeCommand(ArrayList<Task> items, DukeException error, String input) {
+        String[] inputs;
+        switch (command) {
+            case CMD_TODO:
+                addTodo(items, input.replace(TODO, "").trim());
+                addTaskMessage(items.get(taskCount));
+                break;
+            case CMD_EVENT:
+                inputs = input.split(AT);
+                addEvent(items, inputs[0].trim().replace(EVENT, "").trim(), inputs[1].trim());
+                addTaskMessage(items.get(taskCount));
+                break;
+            case CMD_DEADLINE:
+                inputs = input.split(BY);
+                addDeadline(items, inputs[0].trim().replace(DEADLINE, "").trim(), inputs[1].trim());
+                addTaskMessage(items.get(taskCount));
+                break;
+            case CMD_LIST:
+                int j = 1;
+                System.out.println(border);
+                System.out.println("Here are the task in your list:");
+                for (Task item : items) {
+                    if (item != null) {
+                        System.out.print(j + ".");
+                        System.out.println(item);
+                        j++;
+                    }
+                }
+                break;
+            case CMD_DONE:
+                int dividerPosition = input.indexOf(" ") + 1;
+                int endPosition = input.length();
+                if (endPosition > FIVE) {
+                    String num = input.substring(dividerPosition, endPosition);
+                    int taskNum = Integer.parseInt(num) - 1;
+                    items.get(taskNum).markDone();
+                    System.out.println(border + "Nice! task is done " + '\n' + border);
+                }
+                break;
+            case CMD_TERMINATE:
+                printEndMessage();
+                break;
+            case CMD_DELETE:
+                dividerPosition = input.indexOf(" ") + 1;
+                endPosition = input.length();
+                if (endPosition > SEVEN) {
+                    String num = input.substring(dividerPosition, endPosition);
+                    int taskNum = Integer.parseInt(num) - 1;
+                    removeItem(items, taskNum);
+                }
+                break;
+            default:
+                printErrorMessage(error);
+                break;
+        }

```

Comment:
While you have organized different procedure into separate cases, the executeCommand method is still rather long. How about trying [SLAP principle](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard) here?